### DISCUSSION TITLE
Better way for comptime looping through fields and type check

### DISCUSSION THREAD

**Discussion by danieldaeschle** [vlang/v#11800] [Ideas]
```v
$for Type, field in T.fields {
    // Type variable name must be capital and longer than 1 char
    // Type is not accessable from real code
    // Type can be used in comptime calls like $if Type is string
}
```

Part of: https://github.com/vlang/v/issues/43, https://github.com/vlang/v/issues/8346

**Comment by ntrel** [comment]
We also have to think about consistency of field iteration with method iteration, and how we would introspect a parameter of a method. We should be able to introspect each parameter name and each parameter type. I've been thinking about this have come up with quite a long proposal: https://github.com/vlang/v/issues/8440
